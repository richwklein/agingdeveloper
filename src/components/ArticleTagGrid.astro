---
/**
 * @name ArticleTagGrid
 * Component used for rendering a grid of the category and tags associated with
 * a single article.
 *
 * @param {Props} props - The properties of the tag grid.
 * @return {Function} an Astro component factory function.
 *
 * @example
 * <ArticleTagGrid tags={tags} category={category} />
 */
import TagGridLink from '@components/TagGridLink.astro'

interface Props {
  tags: string[]
  category: string
  class?: string
}

const { tags, category, class: className } = Astro.props
---

<nav id="article-tags" class:list={[className]}>
  <ol class:list={['flex list-none flex-wrap gap-1']}>
    <li><TagGridLink name={category} isCategory={true} /></li>
    {
      tags.map((tag) => (
        <li>
          <TagGridLink name={tag} />
        </li>
      ))
    }
  </ol>
</nav>
