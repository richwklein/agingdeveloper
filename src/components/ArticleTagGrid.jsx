import React from "react";
import {Grid} from "@mui/material";
import PropTypes from "prop-types";
import Tag from "./Tag";

const ArticleTagGrid = ({category, tags}) => {
  return (
    <Grid container spacing={1.5}>
      <Grid item key={category}>
        <Tag name={category} isCategory={true} />
      </Grid>
      {tags.map((tag) => {
        return (
          <Grid item key={tag}>
            <Tag name={tag} />
          </Grid>
        );
      })}
    </Grid>
  );
};

ArticleTagGrid.propTypes = {
  category: PropTypes.string.isRequired,
  tags: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default ArticleTagGrid;
