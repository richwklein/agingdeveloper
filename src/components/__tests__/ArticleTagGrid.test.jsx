import React from "react";
import {cleanup, render} from "@testing-library/react";
import ArticleTagGrid from "../ArticleTagGrid";

describe("ArticleTagGrid", () => {
  afterEach(cleanup);

  const tags = ["Theme Park", "Genie+", "Sadness", "I â™¥ Dogs"];

  const renderComponent = () => render(
      <ArticleTagGrid category={"Family"} tags={tags} />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
