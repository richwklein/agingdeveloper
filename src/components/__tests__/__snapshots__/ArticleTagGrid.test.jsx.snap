// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleTagGrid that it renders correctly. 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1.5 css-acctgf-MuiGrid-root"
>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-8g7dbb-MuiTypography-root-MuiLink-root"
      data-link="internal"
      href="/category/family"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-vh8men-MuiSvgIcon-root"
        data-testid="FolderOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m9.17 6 2 2H20v10H4V6h5.17M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
        />
      </svg>
      Family
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-8g7dbb-MuiTypography-root-MuiLink-root"
      data-link="internal"
      href="/tag/theme-park"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-vh8men-MuiSvgIcon-root"
        data-testid="TagOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
        />
      </svg>
      Theme Park
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-8g7dbb-MuiTypography-root-MuiLink-root"
      data-link="internal"
      href="/tag/genie"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-vh8men-MuiSvgIcon-root"
        data-testid="TagOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
        />
      </svg>
      Genie+
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-8g7dbb-MuiTypography-root-MuiLink-root"
      data-link="internal"
      href="/tag/sadness"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-vh8men-MuiSvgIcon-root"
        data-testid="TagOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
        />
      </svg>
      Sadness
    </a>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineHover css-8g7dbb-MuiTypography-root-MuiLink-root"
      data-link="internal"
      href="/tag/i-dogs"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-vh8men-MuiSvgIcon-root"
        data-testid="TagOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 10V8h-4V4h-2v4h-4V4H8v4H4v2h4v4H4v2h4v4h2v-4h4v4h2v-4h4v-2h-4v-4h4zm-6 4h-4v-4h4v4z"
        />
      </svg>
      I â™¥ Dogs
    </a>
  </div>
</div>
`;
